Gems/Atom/Utils/Code/Include/Atom/Utils/StableDynamicStructOfArrays.inl

```diff
@@ -696,82 +762,62 @@ namespace AZ
-            m_destructorCallback = other.m_destructorCallback;
-        }
-        else if (other.m_data)
-        {
-            // If the cast failed, assert here because something is trying to cast between unrelated handle types
-            AZ_Assert(false, "StableDynamicStructOfArraysHandle: Failed to azrtti_cast from %s to %s", OtherType::RTTI_TypeName(), ValueType::RTTI_TypeName());
-
-            // Since we're about to invalidate the other handle, but this handle won't be assuming ownership, free the object referred to by the other handle
-            other.Free();
-        }
-
-        other.Invalidate();
-
-        return *this;
-    }
-
-    template<typename ValueType>
-    void StableDynamicStructOfArraysHandle<ValueType>::Free()
+    template<typename... value_types>
+    inline void StableDynamicStructOfArraysHandle<value_types...>::Free()
     {
         if (IsValid())
         {
             m_destructorCallback(this);
         }
     }
 
-    template<typename ValueType>
-    bool StableDynamicStructOfArraysHandle<ValueType>::IsValid() const
+    template<typename... value_types>
+    inline bool StableDynamicStructOfArraysHandle<value_types...>::IsValid() const
     {
-        return m_data != nullptr;
+        return AZStd::get<0>(m_data) != nullptr;
     }
 
-    template<typename ValueType>
-    bool StableDynamicStructOfArraysHandle<ValueType>::IsNull() const
+    template<typename... value_types>
+    inline bool StableDynamicStructOfArraysHandle<value_types...>::IsNull() const
     {
-        return m_data == nullptr;
+        return AZStd::get<0>(m_data) == nullptr;
     }
 
-    template<typename ValueType>
-    ValueType& StableDynamicStructOfArraysHandle<ValueType>::operator*() const
+    template<typename... value_types>
+    inline typename StableDynamicStructOfArraysHandle<value_types...>::ItemTupleType&
+        StableDynamicStructOfArraysHandle<value_types...>::operator*()
     {
-        return *m_data;
+        return m_data;
     }
 
-    template<typename ValueType>
-    ValueType* StableDynamicStructOfArraysHandle<ValueType>::operator->() const
+    template<typename... value_types>
+    template<size_t RowIndex>
+    inline auto& StableDynamicStructOfArraysHandle<value_types...>::GetItem() const
     {
-        return m_data;
+        return *AZStd::get<RowIndex>(m_data);
     }
 
-    template<typename ValueType>
-    void StableDynamicStructOfArraysHandle<ValueType>::Invalidate()
+    template <typename... value_types>
+    inline void StableDynamicStructOfArraysHandle<value_types...>::Invalidate()
     {
-        m_data = nullptr;
+        AZStd::apply(
+            // We take a reference to the dataItem pointers here, since we're modifying the pointers
+            // themselves not the data they're pointing to
+            [](value_types *&... dataItems)
+            {
+                ((dataItems = nullptr), ...);
+            },
+            m_data
+        );
```

Comment:
nit: technically you only need to set the first pointer to null. Setting them all to null feels a bit cleaner, but it's not as fast 😄  up to you.